openedx/core/djangoapps/user_api/views.py

```diff
@@ -345,10 +350,63 @@ def post(self, request):
             }
             return JsonResponse(errors, status=400)
 
+        password = data.get("password")
+        password_copy = data.get("password_copy")
+        if password and password_copy and password != password_copy:
+            errors = {
+                'password_copy': [{"user_message": "Passwords don't match."}]
+            }
+            return JsonResponse(errors, status=400)
+
         response = JsonResponse({"success": True})
         set_logged_in_cookies(request, response, user)
+
+        if getattr(settings, 'REGISTRATION_EMAIL_FULL_VERIFICATION', None) is not None:
+            email = data.get("email")
+            error = self._verify_email_really_exists(email)
+            if error:
+                errors = {
+                    'email': [{"user_message": error}]
+                }
+                return JsonResponse(errors, status=400)
+
         return response
 
+    def _verify_email_really_exists(self, email):
+        """Check if a email really exists,
+        returns None if success, otherwise an error message
+        """
+        mx_record = None
+        try:
+            domain = email.rsplit('@', 1)[-1]
```

Comment:
@lenacom I agree that we could improve our sign-up form to constrain users from creating a wrong account. However, I think we should do our research before going one way or another, so maybe we can start a separate thread that will address this question.
FYI @schenedx